src/coreclr/tools/aot/ILCompiler.Compiler/Compiler/ObjectWriter/Dwarf/DwarfFde.cs

```diff
@@ -0,0 +1,113 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Diagnostics;
+using System.Buffers;
+using static ILCompiler.ObjectWriter.DwarfNative;
+
+namespace ILCompiler.ObjectWriter
+{
+    internal sealed class DwarfFde
+    {
+        public readonly DwarfCie Cie;
+        public string PcStartSymbolName;
+        public long PcStartSymbolOffset;
+        public ulong PcLength;
+        public string LsdaSymbolName;
+        public string PersonalitySymbolName;
+        public readonly byte[] Instructions;
+
+        public DwarfFde(DwarfCie cie, byte[] instructions)
+        {
+            Cie = cie;
+            Instructions = instructions;
+            PersonalitySymbolName = null;
+        }
+
+        private enum CFI_OPCODE
+        {
+            CFI_ADJUST_CFA_OFFSET,    // Offset is adjusted relative to the current one.
+            CFI_DEF_CFA_REGISTER,     // New register is used to compute CFA
+            CFI_REL_OFFSET,           // Register is saved at offset from the current CFA
+            CFI_DEF_CFA               // Take address from register and add offset to it.
+        }
+
+        /// <summary>
+        /// Convert JIT version of CFI blob into the the DWARF byte code form.
+        /// </summary>
+        public static byte[] CfiCodeToInstructions(DwarfCie cie, byte[] blobData)
+        {
+            int cfaOffset = cie.InitialCFAOffset;
+            var cfiCode = ArrayPool<byte>.Shared.Rent(4096);
```

Comment:
How do we know that 4096 is enough?

Reply:
The JIT blob format only encodes prologs. It’s limited to first 256 byte offsets, limited set of few opcodes, and not saving any register to stack more than once. I can redo the math but it’s essentially impossible for JIT to produce anything longer.

Alternatively, this would be easy to convert to `IBufferWriter<byte>` approach now.

(`DwarfFde` and `DwarfCie` code is the oldest part of the code which is why it used slightly different code style; they were part of an experiment with unwinding data I wrote during NativeAOT/macOS bringup)

Reply:
I see. Maybe worth adding a comment that since the blob covers only for prologs, it can't be long.